### PR TITLE
fix(agents): use OpenCode default colors for native profiles

### PR DESCRIPTION
**Pull request by Maxsky5**
## Summary
- Mirror OpenCode built-in agent color behavior when `app.agents()` omits explicit colors by mapping `ask`, `build`, `docs`, and `plan` to OpenCode CSS vars.
- Define `--icon-agent-*-base` variables in desktop light/dark themes so those semantic colors resolve to the same values as OpenCode.
- Add adapter test coverage to lock native `build`/`plan` fallback colors and prevent regressions.

## Verification
- bun run --filter @openducktor/adapters-opencode-sdk test
- bun run --filter @openducktor/adapters-opencode-sdk typecheck
- bun run --filter @openducktor/adapters-opencode-sdk lint
- bun run --filter @openducktor/adapters-opencode-sdk build
- bun run --filter @openducktor/desktop typecheck
- bun run --filter @openducktor/desktop lint
- bun run --filter @openducktor/desktop build

## Summary by CodeRabbit

* **Style**
  * Added theme color variables for agent icon bases in both light and dark modes.

* **Improvements**
  * Agents without explicit colors now receive sensible default icon colors when applicable; explicit colors are preserved.
  * Agent entries are more strictly validated; malformed or blank entries are filtered out for a more robust catalog.

* **Tests**
  * Added tests covering color resolution, precedence, synthesized defaults, and filtering of invalid agent entries.

### CODE REVIEW

File: apps/desktop/src/styles.css [hunk 1]
```diff
@@ -107,6 +107,11 @@
   --studio-chrome: oklch(0.446 0.043 257.281); /* ~slate-600 */
   --studio-chrome-foreground: oklch(0.968 0.007 247.896); /* ~slate-50 */
 
+  --icon-agent-plan-base: #a753ae;
+  --icon-agent-docs-base: #fcb239;
+  --icon-agent-ask-base: #2090f5;
+  --icon-agent-build-base: #034cff;
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/styles.css]: Thanks for the suggestion. I’m intentionally keeping these specific agent tokens in hex because this PR’s goal is exact parity with OpenCode’s canonical palette values. Converting to oklch introduces representation drift and can subtly diverge from upstream colors.

File: apps/desktop/src/styles.css [hunk 2]
```diff
@@ -210,6 +215,11 @@
   --studio-chrome: oklch(0.332 0.038 260.912); /* ~slate-700 */
   --studio-chrome-foreground: oklch(0.968 0.007 247.896); /* ~slate-50 */
 
+  --icon-agent-plan-base: #edb2f1;
+  --icon-agent-docs-base: #fbb73c;
+  --icon-agent-ask-base: #2090f5;
+  --icon-agent-build-base: #9dbefe;
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/styles.css]: Same rationale here: keeping OpenCode-mirrored palette tokens in canonical hex preserves exact cross-app color parity for these built-in agent profiles.

File: packages/adapters-opencode-sdk/src/catalog-and-mcp.ts [hunk 3]
```diff
@@ -4,6 +4,22 @@ import { asUnknownRecord, readStringProp } from "./guards";
...
+};
+
+const resolveAgentColor = (agentName: string, explicitColor: unknown): string | undefined => {
+  if (typeof explicitColor === "string" && explicitColor.trim().length > 0) {
+    return explicitColor;
+  }
+
+  const normalizedName = agentName.trim().toLowerCase();
+  return OPENCODE_DEFAULT_AGENT_COLORS[normalizedName];
+};
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/adapters-opencode-sdk/src/catalog-and-mcp.ts]: Fixed in 73c2be70: fallback colors now apply only to native agents, and non-native profiles with reserved names no longer receive OpenCode defaults.

File: packages/adapters-opencode-sdk/src/index.test.ts [hunk 4]
```diff
@@ -1689,6 +1689,51 @@ describe("OpencodeSdkAdapter", () => {
...
+          color: "var(--icon-agent-build-base)",
+        }),
+        expect.objectContaining({
+          id: "plan",
+          label: "plan",
+          color: "var(--icon-agent-plan-base)",
+        }),
+      ]),
+    );
+  });
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/adapters-opencode-sdk/src/index.test.ts:1817]: Fixed in 73c2be70: expanded to table-driven coverage for ask/build/docs/plan defaults, and added explicit-color override + non-native reserved-name assertions.

File: packages/adapters-opencode-sdk/src/catalog-and-mcp.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/adapters-opencode-sdk/src/catalog-and-mcp.ts]: Addressed in 73c2be70: resolveAgentColor now guards unknown/malformed names, and fallback mapping is scoped to native profiles only.

File: packages/adapters-opencode-sdk/src/catalog-and-mcp.ts [hunk 5]
```diff
@@ -38,15 +62,23 @@ export const listAvailableModels = async (
...
+
+          const resolvedColor = resolveAgentColor(entry.name, entry.color, entry.native);
+          return {
+            id: entry.name,
+            label: entry.name,
+            ...(entry.description ? { description: entry.description } : {}),
+            mode: entry.mode,
+            ...(entry.hidden !== undefined ? { hidden: entry.hidden } : {}),
+            ...(entry.native !== undefined ? { native: entry.native } : {}),
+            ...(resolvedColor !== undefined ? { color: resolvedColor } : {}),
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/adapters-opencode-sdk/src/catalog-and-mcp.ts]: Fixed in 10c7b4d9: the `app.agents()` mapper now first coerces each item with `asUnknownRecord`, reads `name` via `readStringProp`, rejects null/non-object/whitespace-only names, and only emits valid entries. I also added a regression test covering malformed entries (`null`, non-object, blank name) to ensure they are ignored safely.
